### PR TITLE
Add my station mac.

### PR DESCRIPTION
**Pull request by guoshiuan**
This is the first PR that implements L3 admit API.

### CODE REVIEW

File: dataplane/saiserver/acl.go [hunk 1]
```diff
@@ -292,3 +292,125 @@ func (a *acl) CreateAclEntry(ctx context.Context, req *saipb.CreateAclEntryReque
...
+	}
+	return &saipb.CreateMyMacResponse{Oid: id}, nil
+}
+
+func (m *myMac) RemoveMyMac(ctx context.Context, req *saipb.RemoveMyMacRequest) (*saipb.RemoveMyMacResponse, error) {
+	mi, ok := m.myMacTable[req.GetOid()]
+	if !ok {
+		return nil, status.Errorf(codes.FailedPrecondition, "%d does not exist", req.GetOid())
+	}
+	ed, err := mi.ToEntryDesc(m, false /*with_priority*/)
```
Comment (DanG100) [inline dataplane/saiserver/acl.go:402]: priority is required to delete the entry, remove this and always include it

see https://github.com/openconfig/lemming/blob/main/dataplane/forwarding/fwdtable/flow/flow.go#L236

File: dataplane/saiserver/acl.go [hunk 2]
```diff
@@ -292,3 +292,125 @@ func (a *acl) CreateAclEntry(ctx context.Context, req *saipb.CreateAclEntryReque
...
+	priority       *uint32
+	portID         *uint64 // wildcard if not specified
+	vlanID         *uint32 // wildcard if not specified
+	macAddress     []byte
+	macAddressMask []byte
+}
+
+// ToEntryDesc returns the EntryDesc.
+func (mi *myMacInfo) ToEntryDesc(m *myMac, withPriority bool) (*fwdpb.EntryDesc, error) {
+	ed := &fwdpb.EntryDesc{
```
Comment (DanG100) [inline dataplane/saiserver/acl.go:306]: there is flow entry builder you could use instead https://github.com/openconfig/lemming/blob/main/dataplane/forwarding/fwdconfig/table.go#L167

File: dataplane/saiserver/acl.go [hunk 3]
```diff
@@ -292,3 +292,125 @@ func (a *acl) CreateAclEntry(ctx context.Context, req *saipb.CreateAclEntryReque
...
+	if withPriority {
+		ed.GetFlow().Priority = *mi.priority
+	}
+
+	if mi.vlanID != nil {
+		ed.GetFlow().Fields = append(
+			ed.GetFlow().Fields,
+			&fwdpb.PacketFieldMaskedBytes{
+				FieldId: &fwdpb.PacketFieldId{Field: &fwdpb.PacketField{FieldNum: fwdpb.PacketFieldNum_PACKET_FIELD_NUM_VLAN_TAG}},
+				Bytes:   binary.BigEndian.AppendUint32(nil, *mi.vlanID),
```
Comment (DanG100) [inline dataplane/saiserver/acl.go:327]: vlan tag is a 16 bit field, https://github.com/openconfig/lemming/blob/main/dataplane/forwarding/protocol/attr.go#L94

File: dataplane/saiserver/acl.go [hunk 4]
```diff
@@ -292,3 +292,118 @@ func (a *acl) CreateAclEntry(ctx context.Context, req *saipb.CreateAclEntryReque
 
 	return &saipb.CreateAclEntryResponse{Oid: id}, nil
 }
+
+type myMacInfo struct {
+	priority       *uint32
+	portID         *uint64 // wildcard if not specified
+	vlanID         *uint32 // wildcard if not specified
```
Comment (DanG100) [inline dataplane/saiserver/acl.go:299]: make this uint16, its 32 bits because protobuf hasn't have uint16 types

Comment (guoshiuan) [reply dataplane/saiserver/acl.go:299]: Revised the code.

File: dataplane/saiserver/acl.go [hunk 5]
```diff
@@ -292,3 +292,118 @@ func (a *acl) CreateAclEntry(ctx context.Context, req *saipb.CreateAclEntryReque
...
+func (mi *myMacInfo) ToEntryDesc(m *myMac) (*fwdpb.EntryDesc, error) {
+	fields := []*fwdconfig.PacketFieldMaskedBytesBuilder{
+		fwdconfig.PacketFieldMaskedBytes(fwdpb.PacketFieldNum_PACKET_FIELD_NUM_ETHER_MAC_DST).
+			WithBytes(mi.macAddress, mi.macAddressMask),
+	}
+
+	if mi.vlanID != nil {
+		fields = append(fields,
+			fwdconfig.PacketFieldMaskedBytes(fwdpb.PacketFieldNum_PACKET_FIELD_NUM_VLAN_TAG).
+				WithBytes(binary.BigEndian.AppendUint16(nil, uint16(*mi.vlanID)), binary.BigEndian.AppendUint16(nil, 0x0FFF)))
```
Comment (DanG100) [inline dataplane/saiserver/acl.go:314]: the mask should be 0xFFFF

**Review by DanG100 — APPROVED** [review]

File: dataplane/saiserver/acl.go (hunk 5 above)
Comment (guoshiuan) [reply dataplane/saiserver/acl.go]: VLAN id is 12 bits.

**Pull request merged**